x-pack/plugins/infra/public/alerting/common/components/condition_preview_charts.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonGroup, EuiSelect } from '@elastic/eui';
+import { AutoSizer } from '../../../components/auto_sizer';
+import { idxToAlphabeticalLabel } from '../../../../common/utils/index_to_alphabetical_label';
+
+type Conditions = Array<Record<string, any>>;
+
+const MIN_BUTTON_WIDTH = 42;
+const MIN_LARGE_BUTTON_WIDTH = 110;
+
+export const useConditionSelector = (conditions?: Conditions) => {
+  const [selectedConditionId, setSelectedConditionId] = useState<number>(0);
+  useEffect(() => {
+    if (!conditions) return;
+    if (selectedConditionId > 0 && selectedConditionId > conditions.length - 1) {
+      setSelectedConditionId(Math.max(0, conditions.length - 1));
+    }
+  }, [conditions, selectedConditionId]);
+
+  return [selectedConditionId, setSelectedConditionId] as [
```

Comment:
Is the casting needed here, I find the generic in `useState<number>` usually sorts this out?

Reply:
It's making me cast it because Typescript is interpreting it as `Array<number | SetStateAction>` instead of as a tuple with a strongly typed value for each item. Might be able to fix this if I don't destructure the `useState` on the same line I create it, though.